src/plugin.rs

```diff
@@ -724,142 +834,158 @@ mod tests {
+            for mode in vec![ParseMode::RecordIds, ParseMode::All] {
+                let mut plugin1 = Plugin::new(
+                    GameId::Morrowind,
+                    Path::new("testing-plugins/Morrowind/Data Files/Blank.esm"),
+                );
+                let mut plugin2 = Plugin::new(
+                    GameId::Morrowind,
+                    Path::new("testing-plugins/Morrowind/Data Files/Blank - Master Dependent.esm"),
+                );
 
-            assert_eq!(0, plugin1.overlap_size(&[&plugin2]));
+                assert_eq!(0, plugin1.overlap_size(&[&plugin2]));
 
-            assert!(plugin1.parse_file(false).is_ok());
+                assert!(plugin1.parse_file(mode).is_ok());
 
-            assert_eq!(0, plugin1.overlap_size(&[&plugin2]));
+                assert_eq!(0, plugin1.overlap_size(&[&plugin2]));
 
-            assert!(plugin2.parse_file(false).is_ok());
+                assert!(plugin2.parse_file(mode).is_ok());
 
-            assert_ne!(0, plugin1.overlap_size(&[&plugin2]));
+                assert_ne!(0, plugin1.overlap_size(&[&plugin2]));
+            }
         }
 
         #[test]
         fn overlap_size_should_return_0_when_there_is_no_overlap() {
-            let mut plugin1 = Plugin::new(
-                GameId::Morrowind,
-                Path::new("testing-plugins/Morrowind/Data Files/Blank.esm"),
-            );
-            let mut plugin2 = Plugin::new(
-                GameId::Morrowind,
-                Path::new("testing-plugins/Morrowind/Data Files/Blank.esp"),
-            );
-
-            assert!(plugin1.parse_file(false).is_ok());
-            assert!(plugin2.parse_file(false).is_ok());
-
-            assert!(!plugin1.overlaps_with(&plugin2));
-            assert_eq!(0, plugin1.overlap_size(&[&plugin2]));
+            for mode in vec![ParseMode::RecordIds, ParseMode::All] {
+                let mut plugin1 = Plugin::new(
+                    GameId::Morrowind,
+                    Path::new("testing-plugins/Morrowind/Data Files/Blank.esm"),
+                );
+                let mut plugin2 = Plugin::new(
+                    GameId::Morrowind,
+                    Path::new("testing-plugins/Morrowind/Data Files/Blank.esp"),
+                );
+
+                assert!(plugin1.parse_file(mode).is_ok());
+                assert!(plugin2.parse_file(mode).is_ok());
+
+                assert!(!plugin1.overlaps_with(&plugin2));
+                assert_eq!(0, plugin1.overlap_size(&[&plugin2]));
+            }
         }
 
         #[test]
         fn is_valid_as_light_master_should_always_be_false() {
-            let mut plugin = Plugin::new(
-                GameId::Morrowind,
-                Path::new("testing-plugins/Morrowind/Data Files/Blank - Master Dependent.esm"),
-            );
-            assert!(plugin.parse_file(false).is_ok());
-            assert!(!plugin.is_valid_as_light_master());
+            for mode in vec![ParseMode::RecordIds, ParseMode::All] {
```

Comment:
One problem with this pattern is that if one mode fails, the panic message won't be able to tell you which mode it was. Since there's only two values and only three statements in the loop, I'd consider just unrolling the loop, though it's a trade-off. At some point some sort of parameterised testing framework would become useful.

Also, it doesn't matter, but you could use `&[ParseMode::RecordIds, ParseMode::All]` instead of `vec![ParseMode::RecordIds, ParseMode::All]`.

Reply:
Hmm, that hadn't occurred to me. 

There's a crate called [parameterized](https://crates.io/crates/parameterized), but for some reason it doesn't seem to be working for me. 

[test-case](https://github.com/frondeus/test-case), on the other hand does, though it requires enabling the custom_test_frameworks unstable feature. It's fairly concise: e.g.:

```rust
#[test_case(ParseMode::RecordIds; "when mode is RecordIds")]
#[test_case(ParseMode::All; "when mode is All")]
fn parse_file_should_succeed(mode: ParseMode) {
    let mut plugin = Plugin::new(
         GameId::Morrowind,
        Path::new("testing-plugins/Morrowind/Data Files/Blank.esm"),
    );
 
     assert!(plugin.parse_file(mode).is_ok());
 
     match plugin.data.record_ids {
     RecordIds::NamespacedIds(ids) => assert_eq!(10, ids.len()),
         _ => panic!("Expected namespaced record IDs"),
    }
}
```

produces the test output:
```
test plugin::tests::morrowind::parse_file_should_succeed::when_mode_is_all ... ok                                                                              
test plugin::tests::morrowind::parse_file_should_succeed::when_mode_is_recordids ... ok
```